lib/handleRequest.js

```diff
@@ -110,7 +110,7 @@ function preHandlerCallback (err, request, reply) {
   }
 
   var result = reply.context.handler(request, reply)
-  if (result && typeof result.then === 'function' && !result[kReplySent]) {
+  if (result && typeof result.then === 'function' && !(result instanceof Reply)) {
```

Comment:
You need to check for boh kReplySent and if it is an instance of Reply. 

Also... Is this slowing down our benchmarks?